### PR TITLE
Add tensor damping support to sparse solvers

### PR DESCRIPTION
**Pull request by luisenp**

### CODE REVIEW

File: theseus/extlib/baspacho_solver.cpp
Comment (mhmukadam) [inline theseus/extlib/baspacho_solver.cpp:194]: No similar beta cpu check?

File: theseus/extlib/tests/test_cusolver_lu_solver.py
Comment (mhmukadam) [inline theseus/extlib/tests/test_cusolver_lu_solver.py:27]: Any reason the manual_seed is hash based from params in baspacho test but set to default in lu test?

File: theseus/extlib/baspacho_solver.cpp
Comment (luisenp) [reply theseus/extlib/baspacho_solver.cpp]: Added some macros to clean up all the tensor checks. See 919b8783b030a0be89e4033966b855440482c102.

File: theseus/extlib/tests/test_cusolver_lu_solver.py
Comment (luisenp) [reply theseus/extlib/tests/test_cusolver_lu_solver.py:27]: We could do the same thing here. I was just maintaining consistency with the original code. I'm guessing @maurimo hadn't thought of the hash version before baspacho.

File: theseus/extlib/baspacho_solver.h
Comment (mhmukadam) [inline theseus/extlib/baspacho_solver.h:82]: Hmm alpha/beta here are double while elsewhere are torch::Tensor. Should we add something to the name to reduce ambiguity: alpha_double or alpha_tensor or something? Is this a concern?

File: theseus/optimizer/linear/baspacho_sparse_solver.py
Comment (mhmukadam) [inline theseus/optimizer/linear/baspacho_sparse_solver.py:114]: Was this comment relevant to keep?

File: theseus/optimizer/linear/utils.py
Comment (mhmukadam) [inline theseus/optimizer/linear/utils.py:1]: License header.

Comment (mhmukadam) [reply theseus/optimizer/linear/utils.py:1]: Also add to the new init file in this PR.

**Review by mhmukadam — APPROVED** [review]
Looks great. Nice work on adding this feature to the sparse solvers, especially given the cpp/cuda backends.

File: theseus/extlib/baspacho_solver.h
Comment (luisenp) [reply theseus/extlib/baspacho_solver.h:82]: This is the CUDA kernel for damping, so I think this is fine. It should be the only place where this happens.

**Comment by luisenp** [comment]
@maurimo waiting for you review before merging.

**Review by maurimo — APPROVED** [review]
Looks great to me! Can merge!

**Pull request merged**
